### DISCUSSION TITLE
cannot get the value initialized outside the func

### DISCUSSION THREAD

**Discussion by heliping** [spf13/viper#1196] [Q&A]
when define the globa variable with viper outside the func , it doesnot work.
How to solve the situation?

```
var (
	HOST = config.CONFIG.GetString("kafka.host")
	PORT = config.CONFIG.GetString("kafka.port")
	
)

func SendTask(message string)  {
}
```

**Comment by sagikazarmark** [comment]
Well, defining global variables is a bad practice, avoid it at all cost.

If you can't, consider the order when your values are set in Viper: by calling `GetString` during a global variable definition you essentially try to get a value before anything is set in Viper.

Your example doesn't contain information from which config source you are trying to load config from (flags, file, env var, etc). Make sure to get values from Viper after it's been initialized.
